lib/beaker-hostgenerator/hypervisor/hcloud.rb

```diff
@@ -0,0 +1,28 @@
+require 'beaker-hostgenerator/data'
+require 'beaker-hostgenerator/hypervisor'
+require 'deep_merge/rails_compat'
+
+module BeakerHostGenerator
+  module Hypervisor
+
+    class Hcloud < BeakerHostGenerator::Hypervisor::Interface
+      include BeakerHostGenerator::Data
+
+      def generate_node(node_info, base_config, bhg_version)
+        # split ostype into string and array
+        # ostype is ubuntu2204, debian10, rocky9
+        os, version = node_info['ostype'].split(/(\D+)/).reject!(&:empty?)
+        base_config['image'] = case os
+        when 'ubuntu'
+          puts version
+          "#{os}-#{version[0, 2]}.#{version[2,2]}"
+        when 'centos'
+          [8, 9].include?(os) ? "#{os}-stream-#{version}" : "#{os}-#{version}"
```

Comment:
I'd go the other way around: there's just CentOS 7 (all older versions are EOL) which isn't stream and 10 will likely also include stream.

Reply:
ah okay. I assumed `stream` will be dropped for 10. In that case I will revert the logic.